src/nodes.coffee

```diff
@@ -3710,6 +3720,22 @@ exports.Assign = class Assign extends Base
     if loneObject instanceof Expansion
       loneObject.error 'Destructuring assignment has no target'
 
+  # Show error if there is more than one `Splat`, or `Expansion`.
+  # Examples: [a, b, c..., d, e, f...], [a, b, ..., c, d, ...], [a, b, ..., c, d, e...]
+  disallowMultipleSplats: ->
+    return unless @variable.base instanceof Arr
+    {objects} = @variable.base
+
+    # Count all `Splats`: [a, b, c..., d, e]
+    splats = (i for obj, i in objects when obj instanceof Splat)
+    # Count all `Expansions`: [a, b, ..., c, d]
+    expans = (i for obj, i in objects when obj instanceof Expansion)
+    # Combine splats and expansions.
+    splatsAndExpans = [splats..., expans...]
+    if splatsAndExpans.length > 1
+      # Sort 'splatsAndExpans' so we can show error at first disallowed token.
+      objects[splatsAndExpans.sort()[1]].error "multiple splats/expansions are disallowed in an assignment"
```

Comment:
This is a rather big chunk to copy/paste.

Reply:
Ok DRYed this one up by having it return `{splats, expans, splatsAndExpans}` (and renamed to `getAndCheckSplatsAndExpansions()`)